Add support for external canonical URL (#33)
pelican-plugins/seo: README.md

#### Canonical URL Tag

The SEO Enhancer automatically adds canonical link metadata to `<head>` tags in order to avoid presenting duplicate content to search engines:

> 1. If you have declared the `:external_canonical:` metadata field, the SEO enhancer will use this. It expects a URL, e.g. https://www.example.com/external_canonical_article.html. It can be used if you syndicate content from third parties or want to include your own posts that are already hosted elsewhere. The SEO enhancer will prefer `:external_canonical:` over the `:save_as:` metadata field.

Reviewer: ```suggestion
1. If you have declared the `:external_canonical:` metadata field, the SEO Enhancer will use it. It expects a URL, e.g. `https://www.example.com/external_canonical_article.html`. It can be used if you syndicate content from third parties or want to include your own posts that are already hosted elsewhere. The SEO Enhancer will prefer `:external_canonical:` over the `:save_as:` metadata field.
```

> 2. If you have declared the `:save_as:` metadata field, the SEO enhancer will use this to contruct a canonical link. However if the `:external_canonical:` is set, the `:save_as:` field will be ignored.

Reviewer: ```suggestion
2. If you have declared the `:save_as:` metadata field, the SEO Enhancer will use it to construct a canonical link. However if the `:external_canonical:` is set, the `:save_as:` field will be ignored.
```

3. If both `:external_canonical:` or `:save_as:` metadata fields are absent, it will construct the canonical link using the output filename created.

Example:

```
# article.md
External_Canonical: https://www.example.com/external_canonical_article.html

# article.rst
:external_canonical: https://www.example.com/external_canonical_article.html
```

will result in the following addition to `<head>` tags:

```html
<link rel="canonical" href="https://www.example.com/external_canonical_article.html" />
```

> In all other cases the SEO enhancer will generate the following:

Reviewer: ```suggestion
In all other cases the SEO Enhancer will generate the following:
```

```html
<link rel="canonical" href="<SITEURL>/example.html" />
```

> 

Reviewer: ```suggestion
```

<link rel="canonical" href="<SITEURL>/example.html" />
```

> If you syndicate content from third parties or include your own posts already hosted elsewhere, it is possible to explicitly assign an external canonical link for an article by adding `:canonical:` (rst) or `Canonical:` (Markdown) metadata fields to an article.

Reviewer: Could you explain that the plugin will first check for external canonical, and if not, will fallback on `save_as`, and finally will build the URL from scratch (with `<link rel="canonical" href="<SITEURL>/example.html" />`)?

For Markdown/rst examples, you can reproduce what I did for the section "To populate the :image: field, ...".

Author: Better?

Reviewer: Yes it is!
I just hilighted some typos.
